cpp/src/arrow/ipc/reader.cc

```diff
@@ -391,12 +342,59 @@ class ArrayLoader {
   ArrayData* out_;
 };
 
+Status DecompressBuffers(const std::vector<std::shared_ptr<ArrayData>>& fields,
+                         Compression::type compression, const IpcReadOptions& options) {
+  std::unique_ptr<util::Codec> codec;
+  ARROW_ASSIGN_OR_RAISE(codec, util::Codec::Create(compression));
+
+  auto DecompressOne = [&](int i) {
+    ArrayData* arr = fields[i].get();
+    for (size_t i = 0; i < arr->buffers.size(); ++i) {
+      if (arr->buffers[i] == nullptr) {
+        continue;
+      }
+      if (arr->buffers[i]->size() == 0) {
+        continue;
+      }
+      const uint8_t* data = arr->buffers[i]->data();
+      int64_t compressed_size = arr->buffers[i]->size() - sizeof(int64_t);
+      int64_t uncompressed_size = util::SafeLoadAs<int64_t>(data);
```

Comment:
Question: what about endianness? Should we call `FromLittleEndian` here?

Reply:
I would guess this project is currently would break in many other places on big endian platforms (how does one even test big endian nowadays? On IBM Z mainframes?). I'll add this though

Reply:
Apparently QEMU can emulate big endian, when we have _lots_ of free time we should look into this